dlt/destinations/filesystem/filesystem.py

```diff
@@ -106,6 +82,16 @@ def dataset_path(self) -> str:
         return posixpath.join(self.fs_path, self.config.dataset_name)
 
     def initialize_storage(self, truncate_tables: Iterable[str] = None) -> None:
+        # clean up existing files for tables selected for truncating
+        if truncate_tables and self.fs_client.isdir(self.dataset_path):
+            all_files = self.fs_client.ls(self.dataset_path, detail=False, refresh=True)
+            for table in truncate_tables:
+                search_prefix = posixpath.join(self.dataset_path, f"{self.schema.name}.{table}.")
+                for item in all_files:
```

Comment:
please preserve the comment form the above. it is a nice warning that you cannot trust fsspec` too much:
```
# NOTE: glob implementation in fsspec does not look thread safe, way better is to use ls and then filter
```
and
```
 # NOTE: deleting in chunks on s3 does not raise on access denied, file non existing and probably other errors
```